src/libraries/System.Security.Cryptography/tests/X509Certificates/PfxIterationCountTests.CustomAppDomainDataLimit.cs

```diff
@@ -0,0 +1,140 @@
+                Assert.Contains("2233907", ce.Message);
+            }, name).Dispose();
+        }
+
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [MemberData(memberName: nameof(PfxIterationCountTests.GetCertsWith_IterationCountNotExceedingDefaultLimit_AndNullOrEmptyPassword_MemberData), MemberType = typeof(PfxIterationCountTests))]
+        public void Import_AppDomainDataWithValueZero_IterationCountNotExceedingDefaultLimit_Throws(string name, bool usesPbes2, byte[] blob, long iterationCount)
+        {
+            _ = iterationCount;
+            _ = blob;
+
+            if (usesPbes2 && !PfxTests.Pkcs12PBES2Supported)
+            {
+                throw new SkipTestException(name + " uses PBES2 which is not supported on this version.");
+            }
+
+            RemoteExecutor.Invoke((certName) =>
+            {
+                AppDomain.CurrentDomain.SetData("System.Security.Cryptography.Pkcs12UnspecifiedPasswordIterationLimit", 0);
+
+                PfxInfo pfxInfo = s_certificatesDictionary[certName];
+
+                CryptographicException ce = Assert.Throws<CryptographicException>(() => Import(pfxInfo.Blob));
+                Assert.Contains("2233907", ce.Message);
+            }, name).Dispose();
+        }
+
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [MemberData(memberName: nameof(PfxIterationCountTests.GetCertsWith_IterationCountExceedingDefaultLimit_MemberData), MemberType = typeof(PfxIterationCountTests))]
+        public void Import_AppDomainDataWithValueMinusOne_IterationCountExceedingDefaultLimit(string name, string password, bool usesPbes2, byte[] blob, long iterationCount)
+        {
+            _ = password;
+            _ = blob;
+            _ = iterationCount;
+
+            if (usesPbes2 && !PfxTests.Pkcs12PBES2Supported)
+            {
+                throw new SkipTestException(name + " uses PBES2 which is not supported on this version.");
+            }
+
+            RemoteExecutor.Invoke((certName) =>
+            {
+                AppDomain.CurrentDomain.SetData("System.Security.Cryptography.Pkcs12UnspecifiedPasswordIterationLimit", -1);
+
+                PfxInfo pfxInfo = s_certificatesDictionary[certName];
+
+                if (OperatingSystem.IsWindows())
+                {
+                    // Opting-out with AppDomain data value -1 will still give us error because cert is beyond Windows limit.
+                    // But we will get the CryptoThrowHelper+WindowsCryptographicException.
+                    PfxIterationCountTests.VerifyThrowsCryptoExButDoesNotThrowPfxWithoutPassword(() => Import(pfxInfo.Blob));
+                }
+                else
+                {
+                    Assert.NotNull(Import(pfxInfo.Blob));
+                }
+            }, name).Dispose();
+        }
+
+        public static readonly Dictionary<string, PfxInfo> s_certificatesDictionary
```

Comment:
Style: Fields belong at the top of the class

Reply:
Style: It should either be not public (preferred), or the name should be PascalCased instead of s_camelCased.  (So `CertificateDictionary`, if it's supposed to be visible, for some reason)